src/composables/layers/layers.composable.ts

```diff
@@ -14,13 +14,9 @@ const themes = useThemes()
 export default function useLayers() {
   function hasIntersect(exclusionA: string, exclusionB: string) {
     try {
-      const concat: number[] = JSON.parse(exclusionA)
-        .concat(JSON.parse(exclusionB))
-        .sort((a: number, b: number) => a - b)
+      const concat = JSON.parse(exclusionA).concat(JSON.parse(exclusionB))
 
-      return concat.some(
-        (element, index, array) => index && element === array[index - 1]
-      )
+      return new Set(concat).size < concat.length
```

Comment:
When is this `true`?

Reply:
true when there are duplicates. The exclusion sets of layer 1 and layer 2 are concatenated, if there are elements which are present in layer1 and layer2 the set (without duplicates) is shorter than the array (with duplicates)

Reply:
this function did not work correctly when there were exclusion items as string type, since they could not be ordered correctly.
Before, the logics were:
1- concatenate the exclusion arrays
2- sort numerically
3- check if there are two identical neighbors

I think using the set is easier, and it works with both numbers and strings and a mix of both